src/utils/layout/NodesContext.tsx

```diff
@@ -838,48 +838,53 @@ function isHiddenPage(state: NodesContext, page: LayoutPage | string | undefined
 
 export function isHidden(
   state: NodesContext,
-  nodeOrId: LayoutNode | LayoutPage | undefined | string,
-  nodes: LayoutPages,
+  type: 'page' | 'node',
+  id: string | undefined,
   _options?: IsHiddenOptions,
 ): boolean | undefined {
-  if (!nodeOrId) {
+  if (!id) {
     return undefined;
   }
 
-  if (nodeOrId instanceof LayoutPage) {
-    return isHiddenPage(state, nodeOrId, _options);
+  if (type === 'page') {
+    return isHiddenPage(state, id, _options);
+  }
+
+  const pageKey = state.nodeData[id]?.pageKey;
+  if (pageKey && isHiddenPage(state, pageKey, _options)) {
+    return true;
   }
 
   const options = withDefaults(_options);
   if (options.forcedVisibleByDevTools && options.respectDevTools) {
     return false;
   }
 
-  const id = typeof nodeOrId === 'string' ? nodeOrId : nodeOrId.id;
-  const node = nodes.findById(id);
   const hidden = state.nodeData[id]?.hidden;
-  if (hidden === undefined) {
-    return undefined;
-  }
-
-  if (hidden) {
-    return true;
+  if (hidden === undefined || hidden === true) {
+    return hidden;
   }
 
   if (state.hiddenViaRules[id]) {
     return true;
   }
 
-  const parent = node?.parent;
-  if (parent && parent instanceof BaseLayoutNode && 'isChildHidden' in parent.def && state.nodeData[parent.id]) {
+  const parentId = state.nodeData[id]?.parentId;
+  const parent = parentId ? state.nodeData[parentId] : undefined;
+  const parentDef = parent ? getComponentDef(parent.layout.type) : undefined;
+  if (parent && parentDef && 'isChildHidden' in parentDef) {
     // eslint-disable-next-line @typescript-eslint/no-explicit-any
-    const childHidden = parent.def.isChildHidden(state.nodeData[parent.id] as any, node);
+    const childHidden = parentDef.isChildHidden(parent as any, id);
     if (childHidden) {
       return true;
     }
   }
 
-  return isHidden(state, parent, nodes, options);
+  if (parent) {
+    return isHidden(state, 'node', parent?.layout.id, options);
+  }
```

Comment:
Nitpick, parent cannot be null 😅 

```suggestion
  if (parent) {
    return isHidden(state, 'node', parent.layout.id, options);
  }
```